packages/template-retail-react-app/app/utils/shipment-utils.test.js

```diff
@@ -0,0 +1,379 @@
+    findExistingDeliveryShipment,
+    findExistingPickupShipment,
+    findUnusedDeliveryShipment,
+    areAddressesEqual,
+    cleanAddressForOrder,
+    findDeliveryShipmentWithSameAddress,
+    findDeliveryShipmentWithoutAddress,
+    findShipmentToConsolidate,
+    isDefaultShipmentEmpty
+} from '@salesforce/retail-react-app/app/utils/shipment-utils'
+
+// Mock the address-utils module
+jest.mock('@salesforce/retail-react-app/app/utils/address-utils', () => ({
+    isAddressEmpty: jest.fn()
+}))
+
+// Mock the constants module
+jest.mock('@salesforce/retail-react-app/app/constants', () => ({
+    DEFAULT_SHIPMENT_ID: 'me'
+}))
+
+describe('shipment-utils', () => {
+    let mockBasket
+    let mockIsPickupMethod
+
+    beforeEach(() => {
+        mockBasket = {
+            basketId: 'test-basket',
+            shipments: [
+                {
+                    shipmentId: 'me',
+                    shippingMethod: {id: 'delivery-method'},
+                    shippingAddress: {
+                        address1: '123 Main St',
+                        city: 'Test City',
+                        stateCode: 'CA',
+                        postalCode: '12345',
+                        countryCode: 'US'
+                    }
+                },
+                {
+                    shipmentId: 'shipment-2',
+                    shippingMethod: {id: 'pickup-method'},
+                    c_fromStoreId: 'store-1'
+                },
+                {
+                    shipmentId: 'shipment-3',
+                    shippingMethod: {id: 'delivery-method-2'},
+                    shippingAddress: null
+                }
+            ],
+            productItems: [
+                {productId: 'prod-1', shipmentId: 'me'},
+                {productId: 'prod-2', shipmentId: 'me'},
+                {productId: 'prod-3', shipmentId: 'shipment-2'},
+                {productId: 'prod-4', shipmentId: 'shipment-3'}
+            ]
+        }
+
+        mockIsPickupMethod = jest.fn((shippingMethod) => {
+            return shippingMethod?.id === 'pickup-method'
+        })
+
+        // Mock isAddressEmpty to return false by default
+        const {isAddressEmpty} = jest.requireMock(
+            '@salesforce/retail-react-app/app/utils/address-utils'
+        )
+        isAddressEmpty.mockReturnValue(false)
+    })
+
+    describe('getItemsForShipment', () => {
+        it('should return items for a specific shipment', () => {
+            const items = getItemsForShipment(mockBasket, 'me')
+            expect(items).toHaveLength(2)
+            expect(items[0].productId).toBe('prod-1')
+            expect(items[1].productId).toBe('prod-2')
+        })
+
+        it('should return empty array for non-existent shipment', () => {
+            const items = getItemsForShipment(mockBasket, 'non-existent')
+            expect(items).toHaveLength(0)
+        })
+
+        it('should return empty array for null/undefined inputs', () => {
```

Comment:
nit: please use `test` for consistent with the rest of the code base
```suggestion
        test('should return empty array for null/undefined inputs', () => {
```